rtmpsuck.c

```diff
@@ -776,9 +774,23 @@ TFTYPE doServe(void *arg)	// server socket and state (our listening socket)
     {
       if (!RTMPPacket_IsReady(&ps))
         continue;
-      ServePacket(server, 0, &ps);
+      /* Client can set chunk_size before the connect command */
+      if (ps.m_packetType == RTMP_PACKET_TYPE_CHUNK_SIZE)
+        {
+          if (ps.m_nBodySize >= 4)
+            {
+              server->rs.m_inChunkSize = AMF_DecodeInt32(ps.m_body);
+              RTMP_Log(RTMP_LOGDEBUG, "%s, client: chunk size change to %d", __FUNCTION__,
+                       server->rs.m_inChunkSize);
+            }
+        }
+      else
+        {
+          ServePacket(server, 0, &ps);
+          connect_cmd_processed = TRUE;
```

Comment:
Which client+stream are you using to test this? I feel as though I'm missing something, and I would like to try to understand better what's going on.